aspnetcore/fundamentals/configuration/index.md

```diff
@@ -539,6 +560,38 @@ In the preceding code, `config.AddInMemoryCollection(Dict)` is added after the [
 
 See [Bind an array](#boa) for another example using `MemoryConfigurationProvider`.
 
+::: moniker-end
+::: moniker range=">= aspnetcore-5.0"
+
+<a name="kestrel"></a>
+
+## Kestrel endpoint configuration
+
+Kestrel specific endpoint configuration overrides all [cross-server](xref:fundamentals/servers/index) endpoint configurations. Cross-server endpoint configurations include:
+
+  * [UseUrls](xref:fundamentals/host/web-host#server-urls)
+  * `--urls` on the [command line](xref:fundamentals/configuration/index#command-line)
+  * The [environment variable](xref:fundamentals/configuration/index#environment-variables) `ASPNETCORE_URLS`
+
+Consider the following *appsettings.json* file used in an ASP.NET Core web app:
+
+[!code-json[](~/fundamentals/configuration/index/samples_snippets/5.x/appsettings.json?highlight=2-8)]
+
+When the preceding highlighted markup is used in an ASP.NET Core web app ***and*** the app is launched on the command line with the following cross-server endpoint configuration:
+
+`dotnet run --urls="https://localhost:7777"`
+
+The endpoint configuration from the *appsettings.json* file (`https://localhost:9999`) is used because it's Kestrel specific.
+
+When the Kestrel-specific endpoint is configured as an environment variable, for example:
+
+`set Kestrel__Endpoints__Https__Url=https://localhost:8888`
+
+The environment setting overrides the *appsettings.json* setting and all other configurations.
```

Comment:
Since this is using the EnvironmentVariablesConfigurationProvider and appsettings.json uses the JsonConfigurationProvider, the settings get merged, not overridden which is the standard behavior for IConfigurationProviders. In this case all the config is kestrel-specific so there's no overriding unless multiple providers define endpoints with the same name.

In the case of `Kestrel__Endpoints__Https__Url`, "Https" is the name of the endpoint. If it was `Kestrel__Endpoints__Http__Url`, "Http" is the name of the endpoint. If appsettings defined an endpoint named something else like "Custom", Kestrel would listen on the addresses defined for the "Https", "Http", and "Custom" endpoints.